### DISCUSSION TITLE
repetition field handling

### DISCUSSION THREAD

**Discussion by ddst0111** [microsoft/FHIR-Converter#141] [Q&A: Liquid Engine] [answered]
When trying to parse PID.11 segment, the e.g. the following PID has issue, as it consider 	
"20200819~21 WOOD CT" as one component, it should be repeated field.

 
21 WOOD CT^^HT^NY^11043^US^M^^Harris^^^20200819~21 WOOD CT^^HT^NY^11043^US^H^^Harris^^^20200819~21 WOOD CT^^HT^NY^11043^US^BA^^Harris^^^20200819

this shall be parsed into three addresses, but the convert only converted into one address in FHIR 

 1.   21 WOOD CT^^HT^NY^11043^US^M^^Harris^^^20200819
 2.  21 WOOD CT^^HT^NY^11043^US^H^^Harris^^^20200819
 3.  21 WOOD CT^^HT^NY^11043^US^BA^^Harris^^^20200819

         "address": [
          {
            "line": [
              "21 WOOD CT"
            ],
            "city": "HT",
            "state": "NY",
            "postalCode": "11043",
            "country": "US",
            "type": "postal",
            "period": {
              "start": "20200819~21 WOOD CT",
              "end": "HT"
            }
          }
        ],

Look into the code, under Hl7v2DataParser, shall it first check the repetitions fields then "ParseComponent"?

                    if (!string.IsNullOrEmpty(fieldValues[f]))
                    {
                        var components = ParseComponents(fieldValues[f], encodingCharacters);
                        var field = new Hl7v2Field(fieldValues[f], components);
                        var repetitions = fieldValues[f].Split(encodingCharacters.RepetitionSeparator);
                        for (var r = 0; r < repetitions.Length; ++r)
                        {
                            var repetitionComponents = ParseComponents(repetitions[r], encodingCharacters);
                            var repetition = new Hl7v2Field(repetitions[r], repetitionComponents);
                            field.Repeats.Add(repetition);
                        }

                        fields.Add(field);
                    }
                    else
                    {
                        fields.Add(null);
                    }

**Comment by ACMoretxj** [comment, marked as answer]
This issue was solved in pull request #160 . The FHIR-Converter is able to handle PID.11 and generate multiple address properties.
